Network problem script

rramkumar1 opened the pull request:
Added a script that checks for important networking events. This script will be run by the custom plugin monitor. 

Currently the script only checks if the conntrack table is full and sends a temporary event if it is. More checks will be added in the future.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/node-problem-detector/152)
<!-- Reviewable:end -->

rramkumar1 commented:
/assign @Random-Liu

andyxning commented on config/plugin/network_problem.sh:
```diff
@@ -0,0 +1,15 @@
+
+# This plugin checks for common network issues. Currently, it only checks
+# if the conntrack table is full. 
+
```
I would prefer we keep the plugin pattern for defining exit code. Which is illustrated in [check_ntp.sh](https://github.com/kubernetes/node-problem-detector/blob/master/config/plugin/check_ntp.sh#L6-L8)
```
OK=0
NONOK=1
UNKNOWN=2
```

rramkumar1 replied:
Done

andyxning commented on config/plugin/network_problem.sh:
```diff
@@ -0,0 +1,15 @@
+conntrack_count=$(cat /proc/sys/net/ipv4/netfilter/ip_conntrack_count)
+if (( conntrack_count >= conntrack_max )); then
+  echo "Conntrack table full"
+  exit 1
```
`exit $OK`

rramkumar1 replied:
Done

andyxning commented on config/plugin/network_problem.sh:
```diff
@@ -0,0 +1,15 @@
+fi
+
+echo "Conntrack table available"
+exit 0
```
`exit $OK`

rramkumar1 replied:
Done

andyxning commented on config/plugin/network_problem.sh:
```diff
@@ -0,0 +1,15 @@
+# This plugin checks for common network issues. Currently, it only checks
+# if the conntrack table is full. 
+
+conntrack_max=$(cat /proc/sys/net/ipv4/netfilter/ip_conntrack_max)
```
I think we need to check the file existence firstly, otherwise if one of the two files do not exit, the check script will return error exit code.

rramkumar1 replied:
Done

andyxning commented:
@rramkumar1 @Random-Liu 

I am not sure we need to add all available custom plugins in node-problem-detector repo. But, imo, i am fine with it.

rramkumar1 commented:
@andyxning Addressed your comments, PTAL. Also I agree that it would make sense to add all available custom plugins to the repo since they should be useful for anyone.

andyxning commented:
lgtm

@Random-Liu PTAL.

rramkumar1 commented:
ping @Random-Liu

Random-Liu commented:
LGTM

r4j4h commented on config/plugin/network_problem.sh:
```diff
@@ -0,0 +1,23 @@
+NONOK=1
+UNKNOWN=2
+
+[ -f /proc/sys/net/ipv4/netfilter/ip_conntrack_max ] || echo $UNKNOWN
```
@rramkumar1 @Random-Liu  I see this is merged now and wanted to bring attention to this line just incase :+1:

r4j4h replied:
I created !155 to handle it since I messed this up so poorly!

andyxning replied:
> is echo $UNKNOWN intentional or should it be exit $UNKNOWN?

Apparently it should be `exit $UNKNOWN`. Sorry for not realizing this early.

dchen1107 commented:
30s interval is too often, and I don't think any remedy system will take an action every minute. Can we increase the interval to 5 mins at least? cc/ @Random-Liu
